### DISCUSSION TITLE
Stereochemistry in simulated enzymatic reactions

### DISCUSSION THREAD

**Discussion by mjnoga** [rdkit/rdkit#4679] [Q&A] [answered]
I'm trying to use RDKit reaction functionality to simulate enzymatic reactions. The main goal is to perform reliable tracing of isotope labels across metabolic pathways. 
It appears that current stereochemistry support  in reactions may not be sufficient, or maybe I'm missing something. 

The first example I stumbled upon the problem is aconitase, but as I tested more reactions the issue appears to be universal. 

[Aconitase](https://www.uniprot.org/uniprot/Q99798) Inter-converts citrate and D-Threo-isocitrate and can be encoded in reaction SMARTS that creates two stereocenters:
```python
citrate = Chem.MolFromSmiles("OC(=O)CC(O)(C(=O)O)CC(=O)O")

d_threo_isoctitrate = Chem.MolFromSmiles(
    "O[C@H]([C@H](CC(O)=O)C(O)=O)C(O)=O"
)  # CHEBI:151
display(Chem.MolToSmiles(d_threo_isoctitrate))

aconitase = AllChem.ReactionFromSmarts(
    "[$(CC(=O)O):2][C:3]([O:9])([$(C(=O)O):6])" ">>" "[C@:3]([C:6])[C@@:2][OH:9]",
)
aconitase.Initialize()
products = [p[0] for p in aconitase.RunReactants((citrate,))]
[Chem.SanitizeMol(p) for p in products]
Draw.MolsToGridImage(
    products,
    legends=[Chem.MolToSmiles(p) for p in products],
    subImgSize = (300,300)
)
```
This generates two identical products that are identical with expected product.
<img width="670" alt="Screenshot 2021-11-07 at 22 23 36" src="https://user-images.githubusercontent.com/26528985/140662341-5d1350b7-c066-4fc0-b69c-528cf602774c.png">

However, citric acid molecule is pro-chiral and if it is produced enzymatically by citrate synthase using unlabeled oxaloacetate and acyl-13C2-acetyl-CoA, aconitase will encounter 3S-13C2-citric acid:

```python
citrate_13C2_3S = Chem.MolFromSmiles("O=C(O)C[C@@](O)([13CH2][13C](=O)O)C(=O)O")
isoctitrate_13C2_2R3S = Chem.MolFromSmiles(
    "O[C@H]([C@H]([13CH2][13C](O)=O)C(O)=O)C(O)=O"
)
display(Chem.MolToSmiles(isoctitrate_13C2_2R3S))
products = [p[0] for p in aconitase.RunReactants((citrate_13C2_3S,))]
[Chem.SanitizeMol(p) for p in products]
Draw.MolsToGridImage(
    products,
    legends=[Chem.MolToSmiles(p) for p in products],
    subImgSize = (300,300)
)
```
<img width="677" alt="Screenshot 2021-11-07 at 22 27 16" src="https://user-images.githubusercontent.com/26528985/140662461-6f17362a-9e4a-4f11-ae40-0201d88a7703.png">
Right now there are again two products, where the 1st one is correct, but the 2nd appears to have hydroxyl group moved to wrong carboxymethyl group. 

It appears that code matching substrate SMARTS pattern to substrate structure ignores the stereochemistry. If I use `GetSubstructMatches()` there is `useChirality` keyword argument (disabled by default) that allows correct matching of the substrate pattern: 
```python
ms = citrate_13C2_3S.GetSubstructMatches(
    Chem.MolFromSmarts("[$(CC(=O)O):2][C@:3]([O:9])([$(C(=O)O):6])"),
    useChirality=True,
)
for m in ms:
    citrate_13C2_3S.__sssAtoms = m
    display(citrate_13C2_3S)
```
<img width="422" alt="Screenshot 2021-11-07 at 22 34 14" src="https://user-images.githubusercontent.com/26528985/140662692-e35efdb4-da2c-48c9-8e83-33724a097458.png">
If this keyword is ignored, there are two matches:

```python
ms = citrate_13C2_3S.GetSubstructMatches(
    Chem.MolFromSmarts("[$(CC(=O)O):2][C@:3]([O:9])([$(C(=O)O):6])"),
#     useChirality=True,
)
for m in ms:
    citrate_13C2_3S.__sssAtoms = m
    display(citrate_13C2_3S)
```

<img width="357" alt="Screenshot 2021-11-07 at 22 35 12" src="https://user-images.githubusercontent.com/26528985/140662712-893f993c-b475-432e-9ed1-769e857b6e5b.png">

My question is whether is it possible to enable `useChirality` keyword argument in `rxn.RunReactants()` or get equivalent results by other means?

**Comment by dehaenw** [comment]
In your second picture, these look like they are both the same enantiomer, in one case the hydroxylation occured on the C12 carbon, in the other case on the C13 carbon. If you would convert C13 to C12 they would be the same molecule. I might be picturing it wrong, but can you sketch/gives SMILES of what enantiomer you are expecting to see?

**Reply by dehaenw** [reply]
One thing is somewhat confusing: you can see one product is RS the other one is RR, and yet isotopic differences aside they would be the same enantiomer. However this is explained by the fact that 13C has priority over any 12C when determing configuration, which can be tricky, at least i had to do a double take for that.

**Reply by mjnoga** [reply]
Thanks @dehaenw for your feedback!

The expected stereoisomer of aconitase product with labeled substrate is `'O=C(O)[C@@H]([13CH2][13C](=O)O)[C@@H](O)C(=O)O'` (it is actually shown on top of the picture as the output of `display(Chem.MolToSmiles(isoctitrate_13C2_2R3S))`

Indeed, when you would disregard isotopic labels these molecules are chemically identical. However, in enzymatic reaction environment, which is inherently asymmetrical, seemingly identical molecules behave in different ways. This really confused people a lot when exact mechanism of TCA cycle was elucidated for the first time. There is quite some interesting literature on that, if you are interested you might want to start with this short report: OGSTON, A. G. Interpretation of Experiments on Metabolic Processes, Using Isotopic Tracer Elements. Nature 1948, 162 (4129), 963–963. https://doi.org/10.1038/162963b0. 
(tl;dr version: one of the follow-up enzymatic reactions in the cycle is decarboxylation that produces CO2. In experiments using specifically 14C-labeled substrates people did not see radioactive CO2 showing up and argued this proves the reactions cannot go through citric acid that is a symmetrical intermediate. It turned out that the citrate is not a symmetrical molecule and we all should now learn about pro-chirality in organic chemistry classes.)

Another good point is that the fact 13C has a priority when determining 'absolute' stereo configuration can lead to confusing results. For example, when you use 3,4-13C2-oxaloacetate and 12C2-acetyl-CoA as substrates of citrate synthase you get citrate with R-configuration at position 3. In this case aconitase will move the hydroxyl group to 13C carbon and it will show up as [13C]O2 in the follow up reactions. This is good enough to define how the citrate synthase works (and it is indeed called [Citrate (Si)-synthase.](https://enzyme.expasy.org/EC/2.3.3.1) in enzyme classification). However, as soon you get mixture of labels in substrates (which will happen when you let cells metabolize labeled substrates for long enough) the priority of groups around this stereocenter will keep shifting. I believe that in order to handle this properly one needs to trace how the atoms are relatively located in the 3D space, rather than following R,S- naming convention.

**Reply by dehaenw** [reply]
@mjnoga thanks for the clarification. I had (wrongly) assumed your starting material was the "racemate", so `O=C(O)CC(O)([13CH2][13C](=O)O)C(=O)O` (which would give 50/50 mixture of the two products shown) instead of `O=C(O)C[C@@](O)([13CH2][13C](=O)O)C(=O)O`, which is effectively asymmetric, and as is nicely shown in the figure of the paper you sent, will lead only to one product. However, in this case, there is no true pro-chirality anymore as the starting material is, as you note, already asymmetric due to the non-equivalence of the C13 and C12 substituent. Thanks for the links and information!

**Reply by mjnoga** [reply]
While this is quite interesting case, it can be surely confusing: RDKit stumbles on aconitase reaction selectivity, but in order to explain the case properly you need to talk about why citrate synthase makes assymetric products. 

The concept of pro-chirality is actually only needed for citrate synthase and this actually handled quite well in RDKit, despite some inconsistency with drawing:
``` python
cisy = AllChem.ReactionFromSmarts(
    "S[$(C(=O)):5][C:4]"
    "."
    "[$(CC(=O)O):2][C:3](=[O:9])[$(C(=O)O):6]"
    ">>"
    "O[$(C(=O)):5][C:4][C@@:3]([C:6])([OH:9])[C:2]"
)
cisy.Initialize()
products = [p[0] for p in cisy.RunReactants((acetyl_CoA,oxaloacetate))]
[
    (
        display(Chem.MolToSmiles(p)),
        Chem.SanitizeMol(p),
        display(Chem.MolToSmiles(p)),
        display(Chem.FindMolChiralCenters(
            p, force=True, includeUnassigned=True, useLegacyImplementation=False
        )),
        display(p)
    )
    for p in products
]
```
![image](https://user-images.githubusercontent.com/26528985/141435877-9199dae9-b7a7-410b-abac-dd15983e1f9f.png)

You can see that when just light substrates are used the reaction code correctly recognizes that product is not chiral despite assigning a stereocenter in reaction products, `FindMolChiralCenters()` also works correctly, but somehow hydroxyl group is still depicted with a wedge... 

When you run the same code with labeled acetyl-CoA, the stereocenter is created correctly:

```python
products = [p[0] for p in cisy.RunReactants((_13C2acetyl_CoA,oxaloacetate))]
[
    (
        display(Chem.MolToSmiles(p)),
        Chem.SanitizeMol(p),
        display(Chem.MolToSmiles(p)),
        display(Chem.FindMolChiralCenters(
            p, force=True, includeUnassigned=True, useLegacyImplementation=False
        )),
        display(p)
    )
    for p in products
]
```
![image](https://user-images.githubusercontent.com/26528985/141436150-1d3f5713-8857-49fa-9235-c52984e88147.png)

**Comment by greglandrum** [comment, marked as answer]
Hi @mjnoga : Unfortunately at the moment this isn't possible.
The stereochemistry in the reaction definitions is used solely to define what happens with a stereocenter in the transformation (i.e. is it created/destroyed/preserved/inverted), but not as part of the matching.

I definitely see the use case and agree that it would be useful. It may be possible to add an option to support this, but I will have to do a bit of investigation to be sure.

**Reply by mjnoga** [reply]
Thanks @greglandrum for your clarification! 
At least good to know I'm not missing anything...
I can imagine I could work around this limitation by using `GetSubstructMatches()` to identify atoms and bonds that need to be transformed and then delete/create the bonds myself. I guess it would in the end become some kind of alternative reaction SMARTS parser... 
Could you explain a bit how the reaction functionality is currently implemented? Or point me to the relevant spot in the source code?
